internal/cli/auditing/update.go

```diff
@@ -0,0 +1,126 @@
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package auditing
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/mongodb/mongodb-atlas-cli/atlascli/internal/cli"
+	"github.com/mongodb/mongodb-atlas-cli/atlascli/internal/cli/require"
+	"github.com/mongodb/mongodb-atlas-cli/atlascli/internal/config"
+	"github.com/mongodb/mongodb-atlas-cli/atlascli/internal/file"
+	"github.com/mongodb/mongodb-atlas-cli/atlascli/internal/flag"
+	"github.com/mongodb/mongodb-atlas-cli/atlascli/internal/store"
+	"github.com/mongodb/mongodb-atlas-cli/atlascli/internal/usage"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	atlasv2 "go.mongodb.org/atlas-sdk/v20231115014/admin"
+)
+
+type UpdateOpts struct {
+	cli.GlobalOpts
+	cli.OutputOpts
+	enabled                   bool
+	auditAuthorizationSuccess bool
+	filename                  string
+	auditFilter               string
+	fs                        afero.Fs
+	store                     store.AuditingUpdater
+}
+
+func (opts *UpdateOpts) initStore(ctx context.Context) func() error {
+	return func() error {
+		var err error
+		opts.store, err = store.New(store.AuthenticatedPreset(config.Default()), store.WithContext(ctx))
+		return err
+	}
+}
+
+var updateTemplate = "Auditing configuration successfully updated.\n"
+
+func (opts *UpdateOpts) Run() error {
+	auditLog, err := opts.newAuditLog()
+	if err != nil {
+		return err
+	}
+	r, err := opts.store.UpdateAuditingConfig(opts.ConfigProjectID(), auditLog)
+	if err != nil {
+		return err
+	}
+	return opts.Print(r)
+}
+
+func (opts *UpdateOpts) newAuditLog() (*atlasv2.AuditLog, error) {
+	if opts.filename != "" {
+		fileContent, err := file.LoadFile(opts.fs, opts.filename)
+		if err != nil {
+			return nil, err
+		}
+		opts.auditFilter = string(fileContent)
+	}
+
+	return &atlasv2.AuditLog{
+		Enabled:                   &opts.enabled,
+		AuditAuthorizationSuccess: &opts.auditAuthorizationSuccess,
+		AuditFilter:               &opts.auditFilter,
+	}, nil
+}
+
+// UpdateBuilder atlas auditing update [--file filter.json] [--auditFilter "{"atype": "authenticate"}"] [--auditAuthorizationSuccess] [--enabled] --projectId projectId.
+func UpdateBuilder() *cobra.Command {
+	opts := &UpdateOpts{
+		fs: afero.NewOsFs(),
+	}
+	cmd := &cobra.Command{
+		Use:   "update",
+		Short: "Updates the auditing configuration for the specified project",
+		Long:  fmt.Sprintf(usage.RequiredRole, "Project Owner"),
+		Args:  require.NoArgs,
+		Annotations: map[string]string{
+			"output": updateTemplate,
+		},
+		Example: `  # Audit all authentication events for known users:
+  atlas auditing update --auditFilter "{"atype": "authenticate"}"
```

Comment:
does this work or the quotes have to be escaped?

Reply:
during my testing, I didn't have to escape the quotes

Reply:
in terminal, without escaping `"{"atype": "authenticate"}"` ends in the CLI as `{atype: authenticate}` which is not a valid JSON.
try for example `echo "{"atype": "authenticate"}"`

Also, this format (using single quote) works too: `atlas auditing update --auditFilter '{"atype": "authenticate"}'`

Reply:
Here is an example. The API response had the quote 🤔 . I will update anyway the example to include `'`
```bash
./bin/atlas auditing update --auditFilter "{"atype": "authenticate"}" --enabled -o json  
{
  "auditAuthorizationSuccess": false,
  "auditFilter": "{atype: authenticate}",
  "configurationType": "FILTER_JSON",
  "enabled": true
}
```

Reply:
nvm mi comment I see the answer of the API was wrong indeed 🤦 I am wondering why I didn't get an error

Reply:
here is the reply with `'`. Valid JSON indeed 👍 

./bin/atlas auditing update  --auditFilter '{"atype": "authenticate"}' --enabled -o json
{
  "auditAuthorizationSuccess": false,
  "auditFilter": "{\"atype\": \"authenticate\"}",
  "configurationType": "FILTER_JSON",
  "enabled": true
}

Reply:
updated